CRM/Contact/Form/CustomData.php

```diff
@@ -84,23 +84,18 @@ class CRM_Contact_Form_CustomData extends CRM_Core_Form {
    */
   public function preProcess() {
     $this->_cdType = $_GET['type'] ?? NULL;
-    $this->assign('cdType', FALSE);
+    $isBuildForm = ($_GET['type'] ?? NULL) && CRM_Utils_Request::retrieve('multiRecordDisplay', 'String', $this);
     $this->_multiRecordDisplay = CRM_Utils_Request::retrieve('multiRecordDisplay', 'String', $this);
```

Comment:
It looks like you're doing these things twice (retrieving 'type' from $_GET and 'multiRecordDisplay' from CRM_Utils_Request::retrieve).
Since they've both already been fetched, how about
```suggestion
    $this->_multiRecordDisplay = CRM_Utils_Request::retrieve('multiRecordDisplay', 'String', $this);
    $isBuildForm = $this->_cdType && $this->_multiRecordDisplay;
```

Reply:
yeah sure - I have some misgivings about setting silly properties when they are stored on the form anyway & hence retrievable from the form via retrieve so if things are only set once & never altered I tend to prefer to keep the source intact but am also ok with this change